docs/reference/argocd.md

```diff
@@ -816,10 +816,99 @@ spec:
     scopes: '[groups]'
 ```
 
+### Logs RBAC Enforcement (Argo CD v3.0+)
+
+Starting with Argo CD 3.0, logs RBAC enforcement is enabled by default and logs are treated as a first-class RBAC resource. This means:
+
+1. The `server.rbac.log.enforce.enable` flag has been removed
+2. Logs permissions must be explicitly granted to users/groups/roles
+3. The operator does not manage default RBAC policies - users must define their own policies
+4. Custom roles must explicitly include logs permissions
+
+#### Creating Custom Roles with Logs Access
+
+When creating custom roles, you must explicitly add logs permissions:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  name: example-argocd
+  labels:
+    example: rbac-logs
+spec:
+  rbac:
+    policy: |
+      # Custom role with logs access
+      p, role:custom-role, applications, get, */*, allow
+      p, role:custom-role, logs, get, */*, allow
+      
+      # Assign role to users/groups
+      g, user1, role:custom-role
+```
+
+#### Global Log Viewer Role
+
+You can create a global log viewer role that only has access to logs:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  name: example-argocd
+  labels:
+    example: rbac-log-viewer
+spec:
+  rbac:
+    policy: |
+      # Global log viewer role
+      p, role:global-log-viewer, logs, get, */*, allow
+      
+      # Assign role to users/groups
+      g, log-viewers, role:global-log-viewer
+```
+
+### Migration from Argo CD 2.4 to 3.0
+
+If you're upgrading from Argo CD 2.4 to 3.0, note the following changes:
+
+1. The `server.rbac.log.enforce.enable` flag is no longer supported
+2. Logs RBAC is now enforced by default
+3. Users with existing policies need to explicitly add logs permissions
+4. The operator does not provide default RBAC policies - you must define your own
+
+#### Detection
+
+The following users are **unaffected** by this change:
+- Users who have `server.rbac.log.enforce.enable: "true"` in their `argocd-cm` ConfigMap
+- Users who have `policy.default: role:readonly` or `policy.default: role:admin` in their `argocd-rbac-cm` ConfigMap
+
+The following users are **affected** and should perform remediation:
+- Users who don't have a `policy.default` in their `argocd-rbac-cm` ConfigMap
+- Users who have `server.rbac.log.enforce.enable` set to `false` or don't have this setting at all in their `argocd-cm` ConfigMap
+
+#### Remediation Steps
+
+1. **Quick Remediation:**
+   - Add logs permissions to existing roles
+   - Example: `p, role:existing-role, logs, get, */*, allow`
+
+2. **Recommended Remediation:**
+   - Review existing roles and their permissions
+   - Add logs permissions only to roles that need them
+   - Consider creating a dedicated log viewer role
+   - Define your own RBAC policies as the operator does not provide defaults
+   - Remove the `server.rbac.log.enforce.enable` setting from `argocd-cm` ConfigMap if it was present before the upgrade
+
```

Comment:
Can we move this to a new page titled **Upgrading**?
We have a few other instructions for users to follow before and after migration to ensure a smooth upgrade. Consolidating everything in one place will make it easier for users to follow. 
Additionally, we can use this new page for future upgrade considerations, for example, when we release Operator 1.0, which will include more breaking changes.

Reply:
You need to add a new entry under `nav` to create a new page:
https://github.com/argoproj-labs/argocd-operator/blob/58e52570628facb27fc47cecddd7706987bdc993/mkdocs.yml#L30-L33